File: python/example2_4.py
"""
 Example 2.4
 Data Analysis
"""
%cd C:/Course20/ceR/python

import numpy as np
import pandas as pd
data=pd.read_csv('http://web.pdx.edu/~crkl/ceR/data/longley.txt',sep='\s+',nrows=16)

y=data['EM']
x=data.drop(['EM'],axis=1)
one=pd.Series(np.ones(16))
x=pd.concat([one,x],join='inner',axis=1)

from numpy.linalg import inv, eig, cond
xx = x.T @ x
xy = x.T @ y
invxx = inv(xx)
b = invxx @ xy
b

cm = eig(xx)
cn = np.sqrt(np.max(cm[0])/np.min(cm[0]))
cn
# compare with built-in cond
cond(x)

File: R/example2_4.R
# Example 2.4
# Data Analysis
setwd("C:/Course17/ceR/R")
data<-read.table("http://web.pdx.edu/~crkl/ceR/data/longley.txt",header=T,nrows=16)

y<-as.matrix(data[,7])
#x<-as.matrix(cbind(rep(1,nrow(data)),data[,1:6]))
x<-as.matrix(cbind(1,data[,1:6]))
summary(x)
xx<-t(x)%*%x
xy<-t(x)%*%y
invxx<-solve(xx)
b<-invxx%*%xy
b

cm<-eigen(xx)$val
cn<-sqrt(cm[1]/cm[7])
cond<-kappa(x,exact = TRUE)
cat(cn,cond)  # c(cn,cond)

rm(list=ls())

